lib/Sema/CSSimplify.cpp

```diff
@@ -2070,6 +2070,11 @@ static bool isInPatternMatchingContext(ConstraintLocatorBuilder locator) {
       if (path[lastIdx - 1].is<LocatorPathElt::PatternMatch>() &&
           path[lastIdx].is<LocatorPathElt::FunctionArgument>())
         return true;
+      // `case .foo((let a, _), let b) = question`
+      if (path[lastIdx - 2].is<LocatorPathElt::PatternMatch>() &&
```

Comment:
This check should only happen if `path.size() > 2`

Reply:
Thank you for confirmation.
The original code wasn't good enough for N-level pattern matching, so I'm trying to fix it.